### PR TITLE
refactor(home view): section specs

### PR DESCRIPTION
**Pull request by anandaroop**
This is a follow-up to [this discussion](https://github.com/artsy/metaphysics/pull/6102#pullrequestreview-2355021349) from a couple of weeks ago about re-organizing section specs.

PR contains a big diff, but can be summarized as follows:

- All section instances’ specs are moved to `homeView/sections/__tests__/`
  - One spec per section, named the same as the section
- All section **types** are tested implicitly _through_ their ^corresponding section **instances**
- Specs consist generally of two tests: basic metadata + the resolved connection data

The end result is:

- a reduction in `homeView/__tests__/HomeViewSection.test.ts` from ~1700 lines to ~60 lines, since it now no longer holds all the section specs

- a folder of `__tests__` that matches the sections being tested:
  <pre>
  $ tree src/schema/v2/homeView/sections
  
  <b>src/schema/v2/homeView/sections</b>
  ├── ActiveBids.ts
  ├── CuratorsPicksEmerging.ts
  ├── DiscoverSomethingNew.ts
  ├── ExploreByCategory.ts
  ├── (etc…)
  │ 
  ├── <b>__tests__</b>
  │   ├── ActiveBids.test.ts
  │   ├── CuratorsPicksEmerging.test.ts
  │   ├── DiscoverSomethingNew.test.ts
  │   ├── ExploreByCategory.test.ts
  │   ├── (etc…)
  </pre>

- [documentation](https://github.com/artsy/metaphysics/pull/6133/commits/325660b5f5a5dbc774a284d020a53dd16596b9f7) for this convention

The hope is that this standardization makes it easier for newer contributors  to follow existing examples and keep tests reasonably organized.

### CODE REVIEW

File: docs/adding_a_new_home_view_section.md [hunk 1]
```diff
@@ -150,6 +150,16 @@ This will allow you to query the entire list of home view sections and see your
 
 You should now have a commit that looks something like [this][gh_expose_section_commit] (although the folder structure may be different nowadays).
 
+### 6. Add a test for your new section
```
Comment (olerichter00) [inline docs/adding_a_new_home_view_section.md:153]: 🙏

File: src/schema/v2/homeView/index.ts [hunk 2]
```diff
@@ -60,7 +60,7 @@ export const Section: GraphQLFieldConfig<void, ResolverContext> = {
     }
 
     if (!isSectionDisplayable(section, context)) {
-      throw new Error(`Section requires authorized user: ${id}`)
+      throw new Error(`Section is not displayable: ${id}`)
```
Comment (olerichter00) [inline src/schema/v2/homeView/index.ts:63]: 🙏

**Review by olerichter00 — APPROVED** [review]
This is much better organized 💯

**Review by MounirDhahri — APPROVED** [review]
Impressive work

**Pull request merged**